### PR TITLE
Sign main branch Unified container builds with cosign and perform security scanning

### PR DESCRIPTION
**Pull request by withinfocus**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/VULN-130

## 📔 Objective

Signs Unified container images built off `main` with [Cosign](https://github.com/sigstore/cosign). This uses Sigstore's in-house certificate authority with short-lived keys that are all self-managed with the tool, which will also utilize GitHub's provided OIDC entity. As part of an effort to increase transparency of what we build as an open source company, these signatures are also sent to [Rekor](https://search.sigstore.dev/) -- users of our images are then free to verify the images against that log.

Also throws in container security scanning as that's adjacent in other builds.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/build-unified.yml [hunk 1]
```diff
@@ -142,9 +145,10 @@ jobs:
           repository: bitwarden/server
           token: ${{ steps.retrieve-secret-pat.outputs.github-pat-bitwarden-devops-bot-repo-scope }}
           ref: ${{ steps.server-branch-name.outputs.server_branch }}
-          path: 'server'
+          path: "server"
```
Comment (withinfocus) [inline .github/workflows/build-unified.yml:148]: ⛏️ Lint got this.

File: .github/workflows/build-unified.yml [hunk 2]
```diff
@@ -158,6 +162,36 @@ jobs:
           secrets: |
             "GH_PAT=${{ steps.retrieve-secret-pat.outputs.github-pat-bitwarden-devops-bot-repo-scope }}"
 
+      - name: Install Cosign
+        if: inputs.is_workflow_call == true && github.ref == 'refs/heads/main'
```
Comment (withinfocus) [inline .github/workflows/build-unified.yml:166]: ⚠️ Please check me here. I want this and the step below to only run when this container is built for main branch pushes in `server` -- not sure how feasible that is.

Comment (Eeebru) [reply .github/workflows/build-unified.yml:166]: We haven't been passing this input `is_workflow_call` from the trigger job in server, I will make an update to server. Other than that the `if` check is alright

**Review by Eeebru — APPROVED** [review]

File: .github/workflows/build-unified.yml (hunk 2 above)
Comment (Eeebru) [reply .github/workflows/build-unified.yml:166]: Please hold on merging this so I can merge the change on Server first

Comment (Eeebru) [reply .github/workflows/build-unified.yml:166]: Can be merged now

**Pull request merged**